### DISCUSSION TITLE
About NetX visualization ,and conversion to others

### DISCUSSION THREAD

**Discussion by ahmetakman** [lava-nc/lava-dl#64] [General]
Hello, I would like to suggest a feature. There are a bunch of available visualization modules for the onnx models, pytorch models and so on. And, there are web visualizers like netron.app. If there would be available visualization (or proper conversation) for NetX it would be quite easy to understand what is going on with the network. For example, I am trying to map the nmnist model to lava connected processes so, sometimes it becomes hard to figure out which parameter stands for which parameter when I use .pt file. 

Also, I am curious about what corresponds the in slayer CUBA model to the LIF model in lava ? What does exactly voltage decay mean for a LIF layer?

I am a little bit confused about them so sorry about not-concise questions? 

Objective of issue:

**Lava DL version:**

- [ ] **0.3.0** (feature release)
- [ ] **0.2.1** (bug fixes)
- [x] **0.2.0** (current version)
- [ ] **0.1.2**

**Lava version:**

- [ ] **0.4.0** (feature release)
- [ ] **0.3.1** (bug fixes)
- [x] **0.3.0** (current version)
- [ ] **0.2.0**
- [ ] **0.1.2**

**I'm submitting a ...** 

- [ ] bug report
- [x] feature request
- [x] documentation request

**Current behavior:**

- 

**Expected behavior:**

- 

**Steps to reproduce:**

- 

**Related code:**

```
insert short code snippets here
```

**Other information:**

```
insert the output from lava debug here
```

**Comment by bamsumit** [comment]
You can simply __print__ your network and see what the network is composed of. We don't have immediate plans to have a visualizer for it.

If you want to visualize your network, the easiest supported way is to use `torchviz` to visualize your SLAYER or Bootstrap model.

If you want to visualize your hdf5 `network.net`, you can load it as a SLAYER network using [`slayer.auto`](https://lava-nc.org/lava-lib-dl/slayer/auto.html). __This module is not feature complete yet though.__ There are some implementations missing like for e.g., convolution block and more.

The second part of your question, it's a two-stage conversion of parameters.
* __SLAYER to hdf5__: All the neuron models in SLAYER have a property called `device_params` which is a dictionary of device parameters represented in the hdf5 representation.
* __hdf5 to Lava__: the translation of this is done in [`netx.hdf5.Network.get_neuron_params`](https://lava-nc.org/lava-lib-dl/netx/hdf5.html#lava.lib.dl.netx.hdf5.Network.get_neuron_params)

> For example, I am trying to map the nmnist model to lava connected processes so, sometimes it becomes hard to figure out which parameter stands for which parameter when I use .pt file.

__This is the exact reason for the existence of NetX__ so that the user does not need to go through the parameter conversion process which can get tedious with deep networks and very error prone.

**Comment by ahmetakman** [comment]
Thanks for the explanation. I had been getting random errors and hanging codes for some time so I also tried to convert the weights directly from .pt document. Finally, I have solved the problems I have faced and made it work with the NetX module. But, again thanks for the explanation this would help all the upcoming users.
